prometheus_metrics/src/metrics.rs

```diff
@@ -962,6 +995,16 @@ impl Metrics {
         }
     }
 
+    pub fn set_column_subnet_peers(&self, subnet_id: &str, num_peers: usize) {
+        self.column_subnet_peers
+            .get_metric_with_label_values(&[subnet_id])
+            .expect(
+                "the number of label values should match the number \
+                 of labels that column_subnet_peers was created with",
+            )
+            .set(num_peers as i64)
+    }
```

Comment:
there was one `.expect` previously, now there is 3. Is this something worth crashing the application on? Why not log it as an error instead?

Reply:
```
pub fn set_collection_length(&self, typename: &str, collection_name: &str, value: usize) {
    match self.collection_lengths.get_metric_with_label_values(&[typename, collection_name]) {
        Ok(metric) => metric.set(value as i64),
        Err(e) => {
            // Log the error instead of panicking
            error!(
                "Failed to set collection length for type '{}', collection '{}': {}",
                typename, collection_name, e
            );
        }
    }
}
```